airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/factory/object_storage/ObjectLoaderQueueBeanFactory.kt

```diff
@@ -55,9 +55,13 @@ class ObjectLoaderQueueBeanFactory(
     fun objectLoaderClampedPartSizeBytes(
         @Named("objectLoaderPartQueue") queue: ResourceReservingPartitionedQueue<*>,
         @Named("dataChannelMedium") dataChannelMedium: DataChannelMedium,
+        @Named("dataChannelSocketPaths") dataChannelSocketPaths: List<String>
     ): Long {
         if (dataChannelMedium == DataChannelMedium.SOCKET) {
-            return max(queue.clampedMessageSize, loader.partSizeBytes)
+            return max(
+                (dataChannelSocketPaths.size * 4L * 1024 * 1024),
```

Comment:
Nit:  Move 4 * 1024 * 1024 to a named constant so that the scale is documented (e.g. 4 MB) and why this is used to compute the max of the size.